documentation/1.2/reference/interoperability/osgi.md

```diff
@@ -0,0 +1,134 @@
+- can easily use OSGI standard services
+
+This provides a great and straightforward opportunity to run Ceylon code inside a growing number of JEE 
+application servers or enterprise containers that are base upon (or integrated with) OSGI.
+
+In order to be able to resolve and start Ceylon module archives (`.car` files) inside an OSGI container, you will first need to install, in the OSGI container, all the bundles of the Ceylon distribution and SDK.
+
+#### Retrieving the Ceylon Distribution and SDK for OSGI
+
+In order to be able to resolve and start Ceylon module archives (`.car` files) inside an OSGI container, you will first need to install, in the OSGI container, all the bundles of the Ceylon distribution and SDK.
+
+These bundles are available in a dedicated place on the Ceylon language web site :
+
+- as __OBR bundle repositories__:
+    - Old-style OBR syntax (still used by Apache Felix):
+        
+        https://downloads.ceylon-lang.org/osgi/distribution/1.2.2/repository.xml
+
+        https://downloads.ceylon-lang.org/osgi/sdk/1.2.2/repository.xml
+
+    - New standard R5 OBR syntax:
+        
+        https://downloads.ceylon-lang.org/osgi/distribution/1.2.2/index.xml
+        
+        https://downloads.ceylon-lang.org/osgi/sdk/1.2.2/index.xml
+
+- as __P2 repositories for Eclipse development__ :
+  
+  https://downloads.ceylon-lang.org/osgi/distribution/1.2.2/
+
+  https://downloads.ceylon-lang.org/osgi/sdk/1.2.2/
+
+- as __Zip archives for direct deployment inside containers__ :
+  
+  https://downloads.ceylon-lang.org/osgi/distribution/1.2.2/ceylon.distribution.osgi.bundles-1.2.2.zip
+  
+  https://downloads.ceylon-lang.org/osgi/sdk/1.2.2/ceylon.sdk.osgi.bundles-1.2.2.zip
+
+- as __Apache Karaf (aka JBoss Fuse) [features](http://karaf.apache.org/manual/latest/users-guide/provisioning.html)__:
+    - `ceylon.distribution.runtime`, available in the following feature repository
+    
+        https://downloads.ceylon-lang.org/osgi/distribution/1.2.2/karaf-features.xml
+    
+    - `ceylon.sdk`, available in the following feature repository
+      
+        https://downloads.ceylon-lang.org/osgi/distribution/1.2.2/karaf-features.xml
+
+#### Installing the Ceylon Distribution and SDK in an OSGI container
+
+##### Apache Felix 5.4.0:
+
+1. Let's assume we start with a fresh installation of Apache Felix v5.4.0 (`org.apache.felix.main.distribution-5.4.0.zip`)
+
+2. In the `conf/config.properties` file of the Felix installation directory, add the following property:
+
+        org.osgi.framework.executionenvironment=J2SE-1.7,JavaSE-1.7,J2SE-1.6,JavaSE-1.6,J2SE-1.5,JavaSE-1.5,J2SE-1.4,JavaSE-1.4,J2SE-1.3,JavaSE-1.3,J2SE-1.2,,JavaSE-1.2,CDC-1.1/Foundation-1.1,CDC-1.0/Foundation-1.0,J2ME,OSGi/Minimum-1.1,OSGi/Minimum-1.0
+
+  This is necessary since by default the Felix OSGI container provided execution environments don't include `J2SE-1.7`, which is required by a transitive dependency of the `ceylon.net` module.
```

Comment:
With the Markdown processor that the website uses, this paragraph is not within the list and therefore breaks the list (`3.` below starts at 1 again). Probably needs to be indented a bit more